pkg/providers/aws/provider_postgres.go

```diff
@@ -224,26 +238,56 @@ func (p *PostgresProvider) ReconcilePostgres(ctx context.Context, pg *v1alpha1.P
 		return nil, reconcileStatus, nil
 	}
 
-	pi, err := getRDSInstances(session)
-	if err != nil {
-		// return nil error so this function can be requeued
-		msg := "error getting replication groups"
-		return nil, croType.StatusMessage(msg), err
+	// set updates allowed to false on the CR after successful reconcile
+	if updatesAllowed {
+		pg.Spec.AllowUpdates = false
+		if err := p.Client.Update(ctx, pg); err != nil {
+			return nil, "failed to set updateAllowed to false", err
+		}
 	}
-	// check if the cluster has already been created
-	foundInstance, err := getFoundInstance(pi, rdsCfg)
 
-	updating, message, err := p.rdsApplyStatusUpdate(session, serviceUpdates, foundInstance)
+	return postgres, reconcileStatus, nil
+
+}
+
+func (p *PostgresProvider) checkAndApplyMaintenanceUpdates(pg *v1alpha1.Postgres, rdsSvc rdsiface.RDSAPI, rdsCfg *rds.CreateDBInstanceInput, serviceUpdates *ServiceUpdate) (croType.StatusMessage, error) {
+	logger := p.Logger.WithField("action", "checkAndApplyMaintenanceUpdates")
+	pi, err := getRDSInstances(rdsSvc)
 	if err != nil {
-		errMsg := "failed to service update rds instance"
-		return nil, croType.StatusMessage(errMsg), err
-	}
-	if updating {
-		return nil, message, nil
+		msg := "failed to get rds replication groups"
```

Comment:
seems message text is not correct (copied from Redis), it's not replication groups like in redis (?). Maybe could be like "failed to get Postgres Instances"

Reply:
RDS is from the Amazon Relational Database Service (RDS) and is used for Postgres instances. Redis is deployed by Amazon's ElastiCache service. Even though it's a little confusing it is accurate.

Reply:
In Redis:  "errMsg := "error getting replication groups"
In Postgres:  msg := "failed to get rds replication groups"
Formally All is good. Postgres is RDS, so we know if "rds" appear in the error message - error from Postgres. But from my pov - errors notification are intended to help people understand the issue as soon as possible. I would add Redis/Postgres in this messages (or write RDS), but ok, let's say it's minor, it's up to author and approvers. Maybe if such error happens, it will be clear from other notifications around what is about. Thank you